### PR TITLE
Feature: Add generic Iframe support

### PR DESCRIPTION
**Pull request by Blendman974**
Add the command /iframe.
outline-iframe

Simpler version of #6947 :  
Added a generic `EmbedDescriptor` that matches all links.  

To prevent making all links Iframes (when pasting for example), there is a  `noAutoMatch` boolean (default to false) on the `EmbedDescriptor` that prevents it from being processed as an embed on paste.

Icon is from #5137

### CODE REVIEW

File: shared/editor/embeds/index.tsx [hunk 1]
```diff
@@ -621,6 +624,17 @@ const embeds: EmbedDescriptor[] = [
     icon: <Img src="/images/youtube.png" alt="YouTube" />,
     component: YouTube,
   }),
+  /* The generic iframe embed should always be the last one */
+  new EmbedDescriptor({
+    title: "Iframe",
```
Comment (tommoor) [inline shared/editor/embeds/index.tsx:629]:
```suggestion
    title: "Embed",
```

This seems like better language for the average user, iframe can remain as a search keyword.

File: shared/editor/embeds/index.tsx [hunk 2]
```diff
@@ -55,6 +55,8 @@ export class EmbedDescriptor {
   defaultHidden?: boolean;
   /** Whether the bottom toolbar should be hidden – use this when the embed itself includes a footer */
   hideToolbar?: boolean;
+  /** Whether the embed should be auto matched when pasting a URL */
+  noAutoMatch?: boolean;
```
Comment (tommoor) [inline shared/editor/embeds/index.tsx:59]:
```suggestion
  matchOnPaste?: boolean;
```

Maybe we just get explicit with this?

**Comment by Blendman974** [comment]
Renamed "Iframe" to "Embed" (also changed the asset name for consistency)
And changed the boolean to matchOnPaste. (defaults to true so we don't need to add it on every existing Embeds)
Thx @tommoor

**Comment by rcronin** [comment]
Any idea when this will get merged into the code base? Really excited about this feature

**Comment by tommoor** [comment]
Needs testing and probably will choose another image from our existing libraries, but other than that it's likely to be in v0.79.0

**Comment by airbenich** [comment]
How is the size (width & height) of the embeding calculated? Can the width & height be set manually?

**Comment by Geequlim** [comment]
This feature is simply amazing! Please merge this PR as soon as possible and release a new version—I can't wait any longer!

**Review by tommoor — APPROVED** [review]

**Comment by florian-obradovic** [comment]
Awesome - I get a refused to connect:
CleanShot 2024-09-05 at 21 07 06@2x

**Comment by tommoor** [comment]
In that case Plane is blocking the frame from rendering, there's nothing we can do on Outline side.

**Comment by florian-obradovic** [comment]
@tommoor thanks.
Any suggestions what causes the block?
So, is it a web server setting or in the application code?
It happens with all sites I tried.

**Comment by tommoor** [comment]
If you open developer tools in the browser, the error will be visible in the network tab.

**Pull request merged**